uploaders/common.go

```diff
@@ -0,0 +1,322 @@
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		log.Errorf("Fetching list failed %s", err)
+		return RegisterProvisioningProfileResponseData{}, err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+	logDebugPretty(requestResponse)
+
+	return requestResponse.Data, nil
+}
+
+// UploadProvisioningProfile ...
+func UploadProvisioningProfile(responseData RegisterProvisioningProfileResponseData, outputDirPath string, exportFileName string) error {
+	log.Infof("Upload %s to Bitrise...", exportFileName)
+
+	requestURL := responseData.UploadURL
+
+	files := map[string]string{
+		responseData.UploadFileName: (outputDirPath + "/" + exportFileName),
+	}
+
+	request := createUploadRequest(requestURL, files, Put, map[string]string{})
+
+	// Response struct
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+func createUploadRequest(url string, files map[string]string, requestMethod httpMethod, headers map[string]string) *http.Request {
+	var fContent []byte
+
+	for _, file := range files {
+		f, err := os.Open(file)
+
+		if err != nil {
+			failWithMessage("Failed to read file, err: %s", err)
+		}
+
+		fContent, err = ioutil.ReadAll(f)
+
+		if err != nil {
+			failWithMessage("Failed to read file, err: %s", err)
+		}
+
+	}
+
+	req, err := http.NewRequest(Put.String(), url, bytes.NewReader(fContent))
+
+	if err != nil {
+		failWithMessage("Failed to create upload request, err: %s", err)
+	}
+
+	for key, value := range headers {
+		req.Header.Add(key, value)
+
+	}
+
+	return req
+}
+
+func createRequest(url string, fields map[string]interface{}, requestMethod httpMethod, headers map[string]string) (*http.Request, error) {
+	b := new(bytes.Buffer)
```

Comment:
no need to explicit initialize b: var b bytes.Buffer works great and shows, that we do not care about its initial value.